cluster-autoscaler/core/scaledown/planner/planner.go

```diff
@@ -84,10 +85,12 @@ func New(context *context.AutoscalingContext, processors *processors.Autoscaling
 func (p *Planner) UpdateClusterState(podDestinations, scaleDownCandidates []*apiv1.Node, as scaledown.ActuationStatus, pdb []*policyv1.PodDisruptionBudget, currentTime time.Time) errors.AutoscalerError {
 	p.latestUpdate = currentTime
 	p.actuationStatus = as
-	// TODO: clone cluster snapshot to avoid persisting changes done by the
-	// simulation. Or - better yet - allow the snapshot to be forked twice
-	// and just fork it here.
-	err := p.injectOngoingActuation()
+	err := p.rs.ForkClusterSnapshot()
```

Comment:
`RemovalSimulator` operates on ClusterSnapshot passed through context. Since we have access to `p.context` here, I'd just fork it here instead of extending `removalSimulator` interface. Also note that when we fork, we also need to `defer Revert()` afterwards.

Reply:
Ack, thanks!